app/modules/net_info.c

```diff
@@ -0,0 +1,133 @@
+// ***************************************************************************
+// net_info module for ESP8266 with nodeMCU
+//
+// adopted jan 2017 for commit to nodeMCU by Wolfgang Rosner
+// re-worked by Lukas Voborsky, @voborsky
+// ***************************************************************************
+
+// #define NODE_DEBUG
+ 
+#include "module.h"
+#include "lauxlib.h"
+#include "platform.h"
+#include "mem.h"
+#include "task/task.h"
+
+#include "lwip/ip_addr.h"
+#include "espconn.h"
+#include "lwip/dns.h" 
+#include "lwip/app/ping.h"
+#include "lwip/raw.h"
+
+/* 
+ping_opt needs to be the first element of the structure. It is a workaround to pass the 
+calllback reference to the ping_received function. Pointer the ping_option structure is
+equal to the pointer to net_info_ping_t strcture. Reference to callback function is assumed
+to be just after the ping_option structure.
```

Comment:
The "is assumed" sentence is quite worrying and suggests that there's some pretty evil pointer math going on?

Reply:
This was not precise. I have modified the comment to
```c
/* 
ping_opt needs to be the first element of the structure. It is a workaround to pass the 
calllback reference and self_ref to the ping_received function. Pointer the ping_option 
structure is equal to the pointer to net_info_ping_t structure.
*/
```
The order of other fields is unimportant. `ping_option` just needs to be the first element.